src/AppRouter.tsx

```diff
@@ -0,0 +1,30 @@
+import React, {Component} from "react";
+import {
+    BrowserRouter as Router,
+    Route,
+    Switch
+} from "react-router-dom";
+
+import App from "./frontend/share/components/App";
+import PostPage from "./frontend/posts_page/components/App";
+import ChatRoom from "./frontend/chat_room/components/App";
+import Settings from "./frontend/settings/components/App";
+
+class AppRouter extends Component {
+    render() {
+        return (
+            <Router>
+                <div>
+                    <Switch>
```

Comment:
I think it is better to add switch to the main `App.ts`. That way we don't need to include the sidebars in all these files. And the code will be cleaner.

Reply:
Like 
``` typescript
render() {
    return (
        <left_panel\>
        <switch>
            <...routes ...>
        </switch>
        <right_panel/>);
}
```

Reply:
If I understand it correctly, left_panel and right_panel stuffs refer to the shared side bars and headers right? In that case, the shared components are guaranteed to display on every page.